src/Extensions/ChannelWithNewLines.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Jupyter.Core
+{
+    /// <summary>
+    ///      Provides extension methods used throughout the Jupyter Core library.
+    /// </summary>
+    public static partial class Extensions
+    {
+        /// <summary>
+        ///     Creates a wrapper of an IChannel that adds new lines to every message
+        ///     sent to stdout and stderr.
+        /// </summary>
```

Comment:
It may be good to add a quick `<remarks>` indicating that if `original` is already a `ChannelWithNewLines` that this method returns it unmodified (that's a really good idea, btw!).

Reply:
Thanks, good suggestion! I agree, it's a good idea, but it was already implemented this way in the iqsharp code, so I can't take credit.